supervisord.py

```diff
@@ -47,14 +48,29 @@ def check(self, instance):
             AgentCheck.UNKNOWN: 0
         }
 
-        # Report service checks and uptime for each process
+        proc_regex = instance.get('proc_regex')
         proc_names = instance.get('proc_names')
+        processes = server.supervisor.getAllProcessInfo()
+        monitored_processes = []
+
+        # default to all processes if no process filters are configured
+        if not proc_regex and not proc_names:
+            monitored_processes = processes
+
+        # Add regex matches to process list
+        if proc_regex and len(proc_regex):
+            patterns = '(?:%s)' % '|'.join(proc_regex)
+            for process in processes:
+                if re.match(patterns, process) and process not in monitored_processes:
```

Comment:
Would it be better to just build a list that contains redundant entries then deduplicate it with set()?